### PR TITLE
#6645 Remove Portmapper dependency

### LINKED ISSUE #6645: Remove Portmapper dependency
### Description

Today swirlds-platform-core depends on portmapper library. This brings in a number of transitive dependencies. It would be really nice to remove this dependency.

```
|    |    |    +--- portmapper (2.0.4)	
|    |    |    |    +--- org.apache.commons.lang3 (3.12.0)	
|    |    |    |    +--- [AUTO] org.apache.commons.io (2.11.0)	
|    |    |    |    +--- org.apache.commons.collections4 (4.4)	
|    |    |    |    \--- org.slf4j (2.0.6)
```

It is only used in com.swirlds.p2p.portforwarding.portmapper package which is not used anywhere in our application in my searching.

### PR DESCRIPTION
**Pull request by timo0**
Fix #6645

### CODE REVIEW

**Comment by lpetrovic05** [comment]
also remove SocketConfig.doUpnp

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/Network.java [hunk 1]
```diff
@@ -166,52 +121,80 @@ public static void stopPortForwarding() {
...
-        return result;
+    /**
+     * Determine whether a given IP address is a private address, as defined by RFC 1918.
+     * This works for both IPv4 and IPv6.
+     *
+     * @param addr the IP address to check
+     *
+     * @return true if addr is a private address
+     */
+    private static boolean isPrivateIP(@NonNull InetAddress addr) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/Network.java:165]: Great solution!
But:
I'm missing unit tests for that method. Maybe we create a util class and make the method public in that util class and provide unit tests.

File: build-logic/src/main/kotlin/com.hedera.hashgraph.jpms-modules.gradle.kts [hunk 2]
```diff
@@ -153,10 +153,6 @@ extraJavaModuleInfo {
         exportAllPackages()
         requires("org.connid.framework") // this is missing in POM
     }
-    module("com.offbynull.portmapper:portmapper", "portmapper") {
```
Comment (hendrikebbers) [inline build-logic/src/main/kotlin/com.hedera.hashgraph.jpms-modules.gradle.kts:156]: This is breaking services

**Comment by timo0** [comment]
@lpetrovic05 Removed the property

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/Network.java (hunk 1 above)
Comment (timo0) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/Network.java]: Added

**Review by lpetrovic05 — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Pull request merged**